k8s/plugin/src/upgrade_validations.rs

```diff
@@ -0,0 +1,59 @@
+use crate::constant::SINGLE_REPLICA_VOLUME;
+use anyhow::Error;
+use openapi::apis::Uuid;
+use plugin::rest_wrapper::RestClient;
+
+/// Warning to users before doing an upgrade
+pub(crate) const UPGRADE_WARNING: &str =  "Volumes which make use of a single volume replica instance will be unavailable for some time during upgrade.\nIt is recommended that you do not create new volumes which make use of only one volume replica.";
+
+/// Warning to users before doing an upgrade
+pub(crate) const REBUILD_WARNING: &str =  "The cluster is rebuilding replica of some volumes.\nPlease try after some time or skip this validation by retrying with '--ignore-rebuild` flag.";
+
+/// Warning to users before doing an upgrade
+pub(crate) const SINGLE_REPLICA_VOLUME_WARNING: &str =  "The list below shows the single replica volumes in cluster.\nThese single replica volumes may not be accessible during upgrade.\nPlease retry  with '--skip-single-replica-volume-validation` flag to skip this validation.\n";
+
+#[derive(Debug)]
+pub struct UpgradeValidation {
+    pub rebuild_in_progress_in_cluster: bool,
+    pub single_replica_volumes: Vec<Uuid>,
+}
+
+/// Cluster lever validation before upgrade.
+/// 1. Check for any volume rebuild in progress across the cluster
+/// 2. List all single replica volumes
+pub async fn upgrade_validations() -> Result<UpgradeValidation, Error> {
+    let single_replica_volumes = Vec::new();
+    let mut rebuild_in_progress_in_cluster = false;
+    let mut validation = UpgradeValidation {
+        rebuild_in_progress_in_cluster,
+        single_replica_volumes,
+    };
+    match RestClient::client()
+        .volumes_api()
+        .get_volumes(0, None, None)
```

Comment:
Should we use paging here?

Reply:
I knew only this api . Is there any other api for getting the volumes ?